### DISCUSSION TITLE
Is it possible to render icons on the title bar?

### DISCUSSION THREAD

**Discussion by liumingedwin** [labwc/labwc#1677] [Q&A] [answered]
![20240405_13h36m16s_grim](https://github.com/labwc/labwc/assets/51841740/936bdd81-037c-437a-9066-013d161f5231)
Obviously, the title bar in the left desktop uses a point, instead the icon like Firefox. 
I have know that Firefox rendered it on its own, but how to render icons on all the window?

**Comment by Consolatis** [comment]
For server side decorations in labwc you can only use a static icon (e.g. the same for all windows) instead of the dot.
To archive that you can add a `menu.xbm` file (or `menu-active.png` and `menu-inactive.png` or the same for `.svg`) to your theme directory.

It is not possible to add window specific icons to the labwc decorations.

**Comment by liumingedwin** [comment]
it seems to be a little ugly :(

**Reply by johanmalm** [reply, marked as answer]
I suspect we'll support it one day, but it's not trivial to implement, so requires someone with means+ability to dedicate time. Would include a) significant SSD refactoring, b) icon load/cache/watch, and c) icon theme parsing. For there last point there are non-GTK/Qt libraries (`fdicons` would be my favorite I think), but for the rest it would require implementing.

There is some protocol work going on for this IIRC, but haven't followed it.
